### PR TITLE
Implement #693, fixed some suggestions from #883.

### PR DESCRIPTION
**Pull request by pstasiak**
Implement #693, fixed some suggestions from #883.

### CODE REVIEW

File: JSQMessagesViewController/Model/JSQMessageMediaData.h [hunk 1]
```diff
@@ -78,4 +78,24 @@
...
+ */
+- (NSString *)copyableDataType;
+
+/**
+ *  @return Data object of class corresponding to type returned by `copyableDataType`.
+ *
+ *  @discussion You should return an object that is of a class type appropriate to the representation type,
+ *  which typically is a UTI.
+ */
+- (id)copyableData;
```
Comment (jessesquires) [inline JSQMessagesViewController/Model/JSQMessageMediaData.h]: let's name these new methods:

```objective-c
- (NSString *)mediaDataType;
- (id)mediaData;
```

File: JSQMessagesViewController/Model/JSQMessageMediaData.h [hunk 2]
```diff
@@ -78,4 +78,24 @@
  */
 - (NSUInteger)mediaHash;
 
+@optional
```
Comment (jessesquires) [inline JSQMessagesViewController/Model/JSQMessageMediaData.h]: I think I'd like to target this feature for the 8.0 release, so we can make a breaking change and have this be non-optional instead

**Comment by jessesquires** [comment]
Thanks @pstasiak ! This looks great! :smile: :+1: 

- Left a couple of comments
- Can you update the PR to merge into `release_8.0` instead of develop?

**Comment by jessesquires** [comment]
Oh, also -- if you could squash these commits into a single commit, that would be awesome!

File: JSQMessagesViewController/Model/JSQMessageMediaData.h (hunk 1 above)
Comment (pstasiak) [inline JSQMessagesViewController/Model/JSQMessageMediaData.h]: Sure ;) Renamed.

File: JSQMessagesViewController/Model/JSQMessageMediaData.h (hunk 2 above)
Comment (pstasiak) [inline JSQMessagesViewController/Model/JSQMessageMediaData.h]: Removed.

**Comment by pstasiak** [comment]
> Can you update the PR to merge into release_8.0 instead of develop?
> Oh, also -- if you could squash these commits into a single commit, that would be awesome!
PR with applied changes: #1447
I think we can close this PR in favor of #1447.

**Comment by jessesquires** [comment]
Awesome! Thanks @pstasiak !

**Comment by jeffkibuule** [comment]
Ha, it only took a year to get done. Yay open source!

Apologies I couldn't be the one to do this myself, but I'm glad I laid the foundation.

**Comment by jessesquires** [comment]
Thanks @jeffkibuule ! :smile: 

We didn't merge your code, but we did merge your ideas. :smile:

**Comment by jeffkibuule** [comment]
@jessesquires

Awesome, open idea contribution!

**Pull request closed without merging**